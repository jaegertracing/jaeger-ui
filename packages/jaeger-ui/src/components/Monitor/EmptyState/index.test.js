// Copyright (c) 2021 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { mount } from 'enzyme';
import IoIosCheckmark from 'react-icons/lib/io/ios-checkmark-outline';
import IoIosCloseCircle from 'react-icons/lib/io/ios-circle-outline';
import MonitorATMEmptyState from '.';

const props = {
  configureStatus: false,
  sendDataStatus: false,
};
describe('<MonitorATMEmptyState>not configured', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<MonitorATMEmptyState {...props} />);
  });

  it('does not explode', () => {
    expect(wrapper.length).toBe(1);
  });

  it('shows a loading indicator when loading data', () => {
    expect(wrapper.find(IoIosCloseCircle).length).toBe(2);
  });

  it('ATM snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<MonitorATMEmptyState> partially configured', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <MonitorATMEmptyState
        {...{
          configureStatus: true,
          sendDataStatus: false,
        }}
      />
    );
  });

  it('does not explode', () => {
    expect(wrapper.length).toBe(1);
  });

  it('should render checkbox', () => {
    expect(wrapper.find(IoIosCheckmark).length).toBe(1);
  });

  it('ATM snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
