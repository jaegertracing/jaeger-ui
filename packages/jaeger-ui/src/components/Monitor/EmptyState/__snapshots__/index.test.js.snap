// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonitorATMEmptyState> ATM snapshot test 1`] = `
<Col>
  <Row
    gutter={0}
    justify="center"
  >
    <Col
      offset={8}
      span={8}
    >
      <img
        alt="jaeger-monitor-tab-preview"
        className="monitor-preview-image-empty-state"
        src="monitor.png"
      />
    </Col>
  </Row>
  <Row
    gutter={0}
    justify="center"
  >
    <Col
      className="center-empty-state"
      offset={9}
      span={6}
    >
      <h2
        className="main-title-empty-state"
      >
        Get started with Service Performance Monitoring
      </h2>
      <h3
        className="sub-title-empty-state"
      >
        A high-level monitoring dashboard that helps you cut down the time to identify and resolve anomalies and issues.
      </h3>
      <h4
        className="description-empty-state"
      >
        Service Performance Monitoring aggregates tracing data into RED metrics and visualizes them in service and operation level dashboards.
      </h4>
      <Button
        block={false}
        className="button-empty-state"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
      >
        Read the Documentation
      </Button>
      <Row
        gutter={0}
        justify="center"
      >
        <Col
          offset={2}
          span={20}
        >
          <Alert
            message="Service Performance Monitoring requires a Prometheus-compatible time series database."
            showIcon={true}
            type="info"
          />
        </Col>
      </Row>
    </Col>
  </Row>
</Col>
`;
